### ISSUE TITLE
Problems wit Enumerations

### ISSUE THREAD

**Issue by gkwan-ibm** [OpenLiberty/open-liberty#7172]
@javabean68 commented on [Mon Apr 08 2019](https://github.com/OpenLiberty/guide-rest-client-java/issues/64)

Hi all,

I am experimenting some problems with the management of Java Enums.
It seems that they are being serialized/deserialized in a wrong way by OpenLiberty.

I found a workaround to be able to get it correct but it could well be I didn't configure anything correctly :-)

In http://localhost:9080/openapi/ui/ is generated the following example from the attached yaml:
```json
{
  "title": "1",
  "documentType": "100",
  "report": "paper"
}
```

Submitting this generates a 500 Errror as the Enums cannot be parsed. I suppose the cause is that the Enum's fromValue is not called. **An error is at least in your openapi/ui**

As one instead uses title:_1 documentType: _100 and report PAPER it works...

Even with a couple of Clients (Jersey e.g.) I get the same issues  when I send the POJO  with the Enum Fields set.
Hier can you find the yaml to reproduce the problem:
[openapi.txt](https://github.com/OpenLiberty/guide-rest-client-java/files/3053826/openapi.txt)
and my workaround:
```java
    @Provider
    @Produces("application/json")
     public class SampleMessageBodyWriter implements MessageBodyReader<Sample> {

     @Override
      public boolean isReadable(Class<?> aClass, Type type, Annotation[] annotations, MediaType 
        mediaType) {
        return type == Sample.class;
      }

    /**
     * Jackson kann java 8 OffsetDateTime nicht verwalten.
     * https://stackoverflow.com/questions/27952472/serialize-deserialize-java-8-java-time-with-jackson-json-mapper
     * @param aClass
     * @param type
     * @param annotations
     * @param mediaType
     * @param multivaluedMap
     * @param inputStream
     * @return
     * @throws WebApplicationException
     */
    @Override
    public Sample readFrom(Class<Sample> aClass, Type type, Annotation[] annotations, MediaType mediaType, MultivaluedMap<String, String> multivaluedMap, InputStream inputStream) throws WebApplicationException {
        JsonReader jsonReader = Json.createReader(inputStream);
        JsonObject jsonObject = jsonReader.readObject();

        System.out.println("readFrom: " + jsonObject);

        Sample sample = new Sample();

        try {
            sample.setDocumentType(Sample.DocumentTypeEnum.valueOf(jsonObject.getJsonString( "documentType").getString()));
            sample.setTitle(Sample.TitleEnum.valueOf(jsonObject.getJsonString( "title").getString()));
            sample.setReport(Sample.ReportEnum.valueOf(jsonObject.getJsonString( "report").getString()));
        } catch (IllegalArgumentException e) {
            e.printStackTrace();

            try {
                sample.setDocumentType(Sample.DocumentTypeEnum.fromValue(jsonObject.getJsonString( "documentType").getString()));
                sample.setTitle(Sample.TitleEnum.fromValue(jsonObject.getJsonNumber( "title").toString()));
                sample.setReport(Sample.ReportEnum.fromValue(jsonObject.getJsonString( "report").getString()));
            } catch (Exception e1) {
                e1.printStackTrace();
            }
        }

        return sample;
    }
}
```
Do you have any suggestions? What am I missing here?
Thank you so much in advance,
Regards
Fabio

**Comment by gkwan-ibm** [comment]
@javabean68

**Comment by Azquelt** [comment]
Sorry this hasn't been looked at for so long. You're correct that `mpOpenApi-1.x` does produce the schema for enums incorrectly. It uses the value from `.toString()` whereas the JSON Binding spec requires that deserialization is done using the value from `.name()`.

This problem is fixed in `mpOpenApi-2.0`.
